pkg/cluster/agent_customization_test.go

```diff
@@ -0,0 +1,201 @@
+package cluster
+
+import (
+	"testing"
+
+	v3 "github.com/rancher/rancher/pkg/apis/management.cattle.io/v3"
+	"github.com/rancher/rancher/pkg/settings"
+	"github.com/stretchr/testify/assert"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestAgentCustomization_getAgentCustomization(t *testing.T) {
+	testClusterAgentToleration := []corev1.Toleration{{
+		Effect: "NoSchedule",
+		Key:    "node-role.kubernetes.io/controlplane-test",
+		Value:  "true",
+	},
+	}
+	testClusterAgentAffinity := &corev1.Affinity{
+		NodeAffinity: &corev1.NodeAffinity{
+			PreferredDuringSchedulingIgnoredDuringExecution: []corev1.PreferredSchedulingTerm{
+				{
+					Weight: 1,
+					Preference: corev1.NodeSelectorTerm{
+						MatchExpressions: []corev1.NodeSelectorRequirement{
+							{
+								Key:      "cattle.io/cluster-agent-test",
+								Operator: corev1.NodeSelectorOpIn,
+								Values:   []string{"true"},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+	testClusterAgentResourceReq := &corev1.ResourceRequirements{
+		Limits: map[corev1.ResourceName]resource.Quantity{
+			"cpu":    *resource.NewQuantity(500, resource.DecimalSI),
+			"memory": *resource.NewQuantity(250, resource.DecimalSI),
+		},
+		Requests: map[corev1.ResourceName]resource.Quantity{
+			"cpu":    *resource.NewQuantity(500, resource.DecimalSI),
+			"memory": *resource.NewQuantity(250, resource.DecimalSI),
+		},
+	}
+
+	testFleetAgentToleration := []corev1.Toleration{
+		{
+			Key:      "key",
+			Operator: corev1.TolerationOpEqual,
+			Value:    "value",
+		},
+	}
+	testFleetAgentAffinity := &corev1.Affinity{
+		NodeAffinity: &corev1.NodeAffinity{
+			PreferredDuringSchedulingIgnoredDuringExecution: []corev1.PreferredSchedulingTerm{
+				{
+					Weight: 1,
+					Preference: corev1.NodeSelectorTerm{
+						MatchExpressions: []corev1.NodeSelectorRequirement{
+							{
+								Key:      "fleet.cattle.io/agent",
+								Operator: corev1.NodeSelectorOpIn,
+								Values:   []string{"true"},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+	testFleetAgentResourceReq := &corev1.ResourceRequirements{
+		Limits: corev1.ResourceList{
+			corev1.ResourceCPU: resource.MustParse("1"),
```

Comment:
@slickwarren No, they could theoretically have the same values. I just pulled the fleet resource req values from @manno's test PR since that's what he was using.

Reply:
no preference, but from a readability standpoint this is less obvious what `1` means vs. the cluster agent values. I'm just used to only seeing this in mCPU, so it looked strange to see a 1 here but 500 a few lines before this for cluster agent.